### PR TITLE
[MVT][DO NOT MERGE] Fork to work with the same mvt branch in geoadmin

### PR DESCRIPTION
**Pull request by loicgasser**
@procrastinatio I took your commits from mom_merc, I fixed the search at the front level.

This PR is based on the refactoring of https://github.com/geoadmin/mf-chsdi3/pull/3000

@danduk82 @oterral we now have a new REST endpoints, the same as for KML

`/glstyles` POST a new glstyle
`/glstyles/{id}` UPDATE a glstyle, READ a glstyle, DELETE a glstyle

Accessible in Varnish via 

https://public.dev.bgdi.ch/glstyles/-xpJVSkcRDmTZig9zwdaTA.json

I added the extension in the bucket, I think it's cleaner.
I am deploying this branch to dev and int and will point to this branch in geoadmin fork.

### CODE REVIEW

**Comment by oterral** [comment]
@LoicGasser why not using the same ` files`  endpoint ?

**Comment by oterral** [comment]
@loicgasser can t we remove the extension? it  is always json do we don`t care.

**Comment by oterral** [comment]
and the glstyles files service doesn't use the jsone xtension

File: chsdi/lib/validation/__init__.py [hunk 1]
```diff
@@ -5,6 +5,9 @@
 from chsdi.models.bod import Topics
 
 
+OUTPUT_SRS = ('21781', '2056', '3857', '4326')
```
Comment (danduk82) [inline chsdi/lib/validation/__init__.py]: :+1:

File: chsdi/__init__.py [hunk 2]
```diff
@@ -112,6 +113,20 @@ def main(global_config, **settings):
...
                    methods=['GET', 'POST', 'DELETE'])
 
+    # glstyles json files
+    config.add_route('glstyles_collection', '/glstyles', request_method=('GET', 'POST', 'DELETE'))
+    add_cors_route(config,
+                   '/glstyles',
+                   'glstyles_collection',
+                   headers={'Access-Control-Allow-Credentials': 'true'},
+                   methods=['GET', 'POST', 'DELETE'])
+    config.add_route('glstyles', '/glstyles/{id}', request_method=('GET', 'POST', 'DELETE'))
```
[the remark this answers is not in the document]
Comment (loicgasser) [reply chsdi/__init__.py]: that is the service not the bucket, but ok if you want to change it

File: chsdi/lib/files_handler.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    bucket_file_extension = ''
+    default_mime_type = ''
+    default_route_name = ''
+
+    def __init__(self, request):
+        self.request = request
+
+        # Set up AWS DynamoDB and S3 handlers
+        self.dynamodb_fileshandler = DynamoDBFilesHandler(
+            'geoadmin-file-storage', self.bucket_key_name)
```
[the remark this answers is not in the document]
Comment (loicgasser) [reply chsdi/lib/files_handler.py]: this is the original service.

[the remark this answers is not in the document]
Comment (loicgasser) [reply chsdi/lib/files_handler.py]: for kml:
https://github.com/geoadmin/mf-chsdi3/blob/890478ca90060fa4f40294496125bfa4d323bb49/chsdi/views/files.py

for glstyles:
https://github.com/geoadmin/mf-chsdi3/blob/890478ca90060fa4f40294496125bfa4d323bb49/chsdi/views/glstyles.py

File: chsdi/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (danduk82) [reply chsdi/__init__.py]: Yes please, it should be 1:1 between bucket location and service route

File: chsdi/lib/files_handler.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (danduk82) [reply chsdi/lib/files_handler.py]: what do you mean by "original service"? From what I see now this should be only a metaclass from which both glstyles and kml class inherit, or do I get it wrong?

[the remark this answers is not in the document]
Comment (loicgasser) [reply chsdi/lib/files_handler.py]: oh yes my bad will address that. It should be gone, but this is currently overriden the child classes

File: chsdi/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (loicgasser) [reply chsdi/__init__.py]: ok

**Comment by oterral** [comment]
@danduk, it 's a bit weird to have 2 services that does the same thing, no ?

File: chsdi/lib/validation/__init__.py (hunk 1 above)
Comment (loicgasser) [reply chsdi/lib/validation/__init__.py]: these are Marc's commits. I did not adapt chsdi any further for web mercator. This is why you see so many tests failing.

**Comment by loicgasser** [comment]
@oterral extension remove
@danduk82 glstyles to gl-styles and pointing to the correct table.

**Comment by loicgasser** [comment]
@oterral what makes more sense'

gl-styles/{id} via Varnish and chsdi
or
gl-styles/{id} via Varnish and files/{id} via chsdi with content being the only way to determine if we are dealing with a style or a kml?

**Comment by danduk82** [comment]
@loicgasser shall I cahnge something in varnish for `files/` or `gl-styles`  or does it work already?

**Comment by loicgasser** [comment]
No it works thank you

**Comment by procrastinatio** [comment]
@loicgasser So apparently it is a Web Merc only branch. Who is/was supposed to bring the multi SRID support?

**Comment by loicgasser** [comment]
Swisstopo

**Comment by procrastinatio** [comment]
Can you rebase and fix the failing tests?

**Comment by loicgasser** [comment]
Failing tests come from incomplete support of Web Mercator.

**Comment by danduk82** [comment]
@loicgasser ,
I have merged the old `webmerc` branch into `master` and fixed a few conflicts in this branch. I would like to include the support of MVT in next deploy on 19.12.2018 (the last of the year). What do you think, should we:
- cherry pick your commits here to a new feature branch
- merge this one into `master` ?

**Comment by danduk82** [comment]
I will do some cherry-picks on Monday ;)

**Pull request closed without merging**